### PR TITLE
convert viewlogs to vue

### PR DESCRIPTION
**Pull request by OmgImAlexis**
This more or less converts the whole page over. There are a few places that're using Python vars that I'd like to replace in the review.

### CODE REVIEW

File: themes-default/slim/views/viewlogs.mako [hunk 1]
```diff
@@ -64,74 +122,61 @@ pre {
...
+            </div>
+        </div>
+    </div> <!-- End form group -->
 </div> <!-- row -->
 <div class="row">
-    <div class="col-md-12 ${'fanartOpacity' if app.FANART_BACKGROUND else ''}">
-        <pre><div class="notepad"><app-link id="viewlog-text-view" href="errorlogs/viewlog/?text_view=1"><img src="images/notepad.png"/></app-link></div>${log_lines}</pre>
+    <div :class="{fanartOpacity: fanartOpacity}" class="col-md-12">
+        <div class="notepad"><span @click="viewLogAsText"><img src="images/notepad.png"/></span></div>
+        <pre v-html="logs">${log_lines}</pre>
```
Comment (sharkykh) [inline themes-default/slim/views/viewlogs.mako]: Took me way too long to figure out this issue. Vue was trying to parse `{{PROPER|REPACK|REAL|RERIP}}` in the logs.
Please remove the Mako expression.

File: themes-default/slim/views/viewlogs.mako [hunk 2]
```diff
@@ -15,43 +30,86 @@ const startVue = () => {
...
+            },
+            async getLogs() {
+                const { params } = this;
+
+                this.disabled = true;
+
+                const data = await $.get('errorlogs/viewlog/?' + params);
+                history.pushState('data', '', 'errorlogs/viewlog/?' + params);
+
+                this.logs = $(data).find('pre').html();
```
Comment (OmgImAlexis) [inline themes-default/slim/views/viewlogs.mako]: This makes https://github.com/pymedusa/Medusa/pull/4104/commits/df69d36f2e2228781a66c1d25304ee0d88e9b133 unusable.

File: themes-default/slim/views/viewlogs.mako [hunk 3]
```diff
@@ -15,43 +30,86 @@ const startVue = () => {
...
                 const win = window.open('errorlogs/viewlog/?' + params + '&text_view=1', '_blank');
                 win.focus();
-            });
+            },
+            async getLogs() {
+                const { params } = this;
+
+                this.disabled = true;
+
+                const data = await $.get('errorlogs/viewlog/?' + params);
```
Comment (sharkykh) [inline themes-default/slim/views/viewlogs.mako]: Need to change this to get the data from `/api/v2/log`.

File: themes-default/slim/views/viewlogs.mako [hunk 4]
```diff
@@ -15,43 +30,86 @@ const startVue = () => {
...
+                history.pushState('data', '', 'errorlogs/viewlog/?' + params);
```
Comment (sharkykh) [inline themes-default/slim/views/viewlogs.mako]: Don't know what about the browser history. I assume we can still do this even if we're pulling data from the API.

File: themes-default/slim/views/viewlogs.mako [hunk 5]
```diff
@@ -77,10 +77,14 @@ const startVue = () => {
                     min_level: minLevel,
                     log_filter: filter,
                     log_period: period,
-                    log_search: search
+                    log_search: search,
+                    limit: 100
```
Comment (OmgImAlexis) [inline themes-default/slim/views/viewlogs.mako]: @sharkykh we need a way to pass a time as a limit instead of just a count of items.

**Comment by sharkykh** [comment]
Superseded by #6738

**Pull request closed without merging**